config/iam/config.go

```diff
@@ -0,0 +1,234 @@
+limitations under the License.
+*/
+
+package iam
+
+import (
+	"github.com/crossplane-contrib/terrajet/pkg/config"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	//iamv1alpha1 "github.com/crossplane-contrib/provider-tf-aws/apis/iam/v1alpha1"
+)
+
+var SelfPkgPath = "github.com/crossplane-contrib/provider-tf-aws/config/iam"
+
+func ExternalNameAsName(base map[string]interface{}, externalName string) {
+	base["name"] = externalName
+}
+
+func PolicyARNExtractor() reference.ExtractValueFn {
+	return func(mg resource.Managed) string {
+		/*		r, ok := mg.(*iamv1alpha1.Policy)
+				if !ok {
+					return ""
+				}
+				return reference.FromPtrValue(r.Status.AtProvider.Arn)*/
+		paved, err := fieldpath.PaveObject(mg)
+		if err != nil {
+			// todo(hasan): should we log this error?
+			return ""
+		}
+		r, err := paved.GetString("status.atProvider.arn")
+		if err != nil {
+			// todo(hasan): should we log this error?
+			return ""
+		}
+		return r
+	}
+}
+
+func init() {
+	config.Store.SetForResource("aws_iam_access_key", config.Resource{
+		ExternalName: config.ExternalName{
+			DisableNameInitializer: true,
+		},
+		References: config.References{
+			"user": config.Reference{
+				Type: "User",
+			},
+		},
+	})
+
+	config.Store.SetForResource("aws_iam_group", config.Resource{
+		// Cannot use kind "Group" since it conflict with generated constant
+		// name in group version info
+		Kind: "IAMGroup",
+		ExternalName: config.ExternalName{
+			ConfigureFunctionPath: SelfPkgPath + ".ExternalNameAsName",
+			OmittedFields: []string{
+				"name",
+			},
+		},
+	})
+
+	config.Store.SetForResource("aws_iam_group_policy", config.Resource{
+		ExternalName: config.ExternalName{
+			ConfigureFunctionPath: SelfPkgPath + ".ExternalNameAsName",
+			OmittedFields: []string{
+				"name",
+				"name_prefix",
+			},
+		},
+		References: config.References{
+			"group": config.Reference{
+				Type: "IAMGroup",
+			},
+		},
+	})
+
+	config.Store.SetForResource("aws_iam_group_policy_attachment", config.Resource{
+		ExternalName: config.ExternalName{
+			DisableNameInitializer: true,
+		},
+		References: config.References{
+			"group": config.Reference{
+				Type: "IAMGroup",
+			},
+			"policy_arn": config.Reference{
+				Type:      "Policy",
+				Extractor: SelfPkgPath + ".PolicyARNExtractor()",
+			},
+		},
+	})
+
+	config.Store.SetForResource("aws_iam_instance_profile", config.Resource{
+		ExternalName: config.ExternalName{
+			ConfigureFunctionPath: SelfPkgPath + ".ExternalNameAsName",
+			OmittedFields: []string{
+				"name",
+				"name_prefix",
+			},
+		},
+		References: config.References{
+			"role": config.Reference{
+				Type: "Role",
+			},
+		},
+	})
+
+	config.Store.SetForResource("aws_iam_policy", config.Resource{
+		ExternalName: config.ExternalName{
+			DisableNameInitializer: true,
```

Comment:
I'm guessing that this is because it uses ARN as its ID. Could you add a comment regarding this?